src/main/java/de/unijena/cheminf/mortar/model/io/DynamicSMILESFileReader.java

```diff
@@ -0,0 +1,266 @@
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ *
+ *
+ * @author Jonas Schaub, Samuel Behr
+ */
+public class DynamicSMILESFileReader {
+    /**
+     * Possible SMILES file separators used to separate SMILES code from ID. Ordered so that non-whitespace characters
+     * are tested first.
+     */
+    public static final String[] POSSIBLE_SMILES_FILE_SEPARATORS = {"\n", ",", ";", " ", "\t"};
+    //
+    /**
+     * Maximum number of lines starting from the first one to check for valid SMILES strings in a SMILES file when trying to determine
+     * the SMILES code column and separator.
+     */
+    public static final int MAXIMUM_LINE_NUMBER_TO_CHECK_IN_SMILES_FILES = 10;
+    //
+    /**
+     *
+     */
+    public static final List<String> PARSABLE_STRING_EXCEPTIONS = new ArrayList<>(List.of(new String[]{"ID"}));
+    //
+    /**
+     *
+     */
+    private static final Logger LOGGER = Logger.getLogger(DynamicSMILESFileReader.class.getName());
+    //
+    private DynamicSMILESFileReader() {
+
+    }
+    //
+    /**
+     * Checking for parsable SMILES code and saving the determined separator character and SMILES code and ID column positions.
+     *
+     * @param aFile
+     * @return
+     * @throws IOException
+     */
+    public static DynamicSMILESFileFormat detectFormat(File aFile) throws IOException {
+        try (
+                FileReader tmpSmilesFileReader = new FileReader(aFile);
+                BufferedReader tmpSmilesFileBufferedReader = new BufferedReader(tmpSmilesFileReader, BasicDefinitions.BUFFER_SIZE)
+        ) {
+            IChemObjectBuilder tmpBuilder = SilentChemObjectBuilder.getInstance();
+            // AtomContainer to save the parsed SMILES in
+            IAtomContainer tmpMolecule = tmpBuilder.newAtomContainer();
+            SmilesParser tmpSmilesParser = new SmilesParser(tmpBuilder);
+            String tmpSmilesFileDeterminedSeparator = null;
+            int tmpSmilesCodeExpectedPosition = -1;
+            int tmpIDExpectedPosition = -1;
+            int tmpCurrentLineInFileCounter = -1;
+            String tmpSmilesFileCurrentLine = tmpSmilesFileBufferedReader.readLine();
+            //current line read is 0th line
+            tmpCurrentLineInFileCounter++;
+            // as potential headline, the first (0th) line should be avoided for separator determination
+            String tmpSmilesFileFirstLine = tmpSmilesFileCurrentLine;
+            findSeparatorLoop:
+            while (!Thread.currentThread().isInterrupted() && tmpCurrentLineInFileCounter < DynamicSMILESFileReader.MAXIMUM_LINE_NUMBER_TO_CHECK_IN_SMILES_FILES) {
+                tmpSmilesFileCurrentLine = tmpSmilesFileBufferedReader.readLine();
```

Comment:
Because it might be a headline not containing a SMILES code. But we can actually try to parse it anyway. Fixed in https://github.com/FelixBaensch/MORTAR/pull/31/commits/cd86c88b39b82dcdda61f315fe3e686e1990bbea

Reply:
Thank you